src/In.ProjectEKA.HipService/DataFlow/DataFlow.cs

```diff
@@ -0,0 +1,39 @@
+namespace In.ProjectEKA.HipService.DataFlow
+{
+    using System;
+    using System.Threading.Tasks;
+    using HipLibrary.Patient.Model.Response;
+    
+    public class DataFlow: IDataFlow
+    {
+        private readonly IDataFlowRepository dataFlowRepository;
+
+        public DataFlow(IDataFlowRepository dataFlowRepository)
+        {
+            this.dataFlowRepository = dataFlowRepository;
+        }
+        
+        public async Task<Tuple<HealthInformationResponse, ErrorResponse>> HealthInformationRequestFor(HealthInformationRequest request)
+        {
+            if (!ValidateConsentArtefact(request.Consent))
+            {
+                return new Tuple<HealthInformationResponse, ErrorResponse>(null, null);
+            }
+            
+            var result = await dataFlowRepository.SaveRequestFor(request.TransactionId, request)
```

Comment:
If the transactionId is not unique, will there be an exception? If so, where are we handling it?

Reply:
Yeah it is handled in repositry